### PR TITLE
endless-uefi-grub.cfg: Support combined live + installer boot

### PR DESCRIPTION
**Pull request by carlocaione**
https://phabricator.endlessm.com/T12512

### CODE REVIEW

File: debian/endless-uefi-grub.cfg [hunk 1]
```diff
@@ -1,4 +1,15 @@
-search --set=root --fs-type --hint $bootdev 4f68bce3-e8cd-4db1-96e7fbcaf984b709
-probe -u $root --set=rootuuid
-export rootuuid
+set first_part=($bootdev,gpt1)
+
+if [ -e $first_part/eos/image ]; then
+	loopback loop0 $first_part/eos/image
+	set root=(loop0,gpt3)
+	probe -u $root --set=image_dev_uuid
```
Comment (dsd) [inline debian/endless-uefi-grub.cfg]: Is this definitely reading the GPT partition UUID rather than the fs uuid?

Comment (wjt) [inline debian/endless-uefi-grub.cfg]: The `probe` builtin seems undocumented but [this](https://github.com/endlessm/grub/blob/master/grub-core/commands/probe.c#L46) and [this](https://github.com/endlessm/grub/blob/master/grub-core/commands/probe.c#L113-L133) do suggest it's the filesystem's UUID.

Comment (wjt) [inline debian/endless-uefi-grub.cfg]: I've tested this locally and it fetches the FS UUID (as printed by `dumpe2fs -h`) not the GPT UUID (as printed by `sfdisk --part-uuid`).

File: debian/endless-uefi-grub.cfg [hunk 2]
```diff
@@ -1,4 +1,15 @@
...
+	set kparams="endless.image.device=PARTUUID=$image_dev_uuid endless.image.path=/eos/image endless.live_boot"
```
Comment (wjt) [inline debian/endless-uefi-grub.cfg]: In the boot-from-NTFS case, we won't want to pass `endless.live_boot`. Maybe we drop a flag file alongside the image to indicate one way or the other? (I suppose that's a bit fragile.)

File: debian/endless-uefi-grub.cfg [hunk 3]
```diff
@@ -1,4 +1,15 @@
...
+	export kparams
+else
+	search --set=root --fs-type --hint $bootdev 4f68bce3-e8cd-4db1-96e7fbcaf984b709
+	probe -u $root --set=rootuuid
+	export rootuuid
+fi
+
 configfile ($root)/boot/grub/grub.cfg
```
Comment (wjt) [inline debian/endless-uefi-grub.cfg]: Presumably, this path will need to change in the image-in-exfat case to `/eos/boot/...` or similar, and we'll need some way to keep it up-to-date in the image-in-NTFS case.

File: debian/endless-uefi-grub.cfg (hunk 1 above)
Comment (dsd) [inline debian/endless-uefi-grub.cfg]: It's probably fine to use the filesystem UUID here, it should be just as unique, I will add support for this on the initramfs side.

File: debian/endless-uefi-grub.cfg (hunk 2 above)
Comment (dsd) [inline debian/endless-uefi-grub.cfg]: BTW when I opened this task for Carlo it was only considering the live-USB boot case, and only UEFI. For MBR, and for boot-from-windows-install, we had too many open questions to be able to specify what needs to be done.

File: debian/endless-uefi-grub.cfg [hunk 4]
```diff
@@ -1,4 +1,18 @@
-search --set=root --fs-type --hint $bootdev 4f68bce3-e8cd-4db1-96e7fbcaf984b709
-probe -u $root --set=rootuuid
-export rootuuid
-configfile ($root)/boot/grub/grub.cfg
+search.file /endless/endless.img eoslive $bootdev,
+
+# Can't limit search.file to one device; so fake it:
+if [ ( "$eoslive" -a "$eoslive" ">=" "$bootdev," -a $eoslive "<" "$bootdev-" ) ]; then
```
Comment (carlocaione) [inline debian/endless-uefi-grub.cfg]: we need to escape the second $eoslive also?

Comment (wjt) [inline debian/endless-uefi-grub.cfg]: I suppose so, good catch.

**Comment by wjt** [comment]
"LGTM" from @carlocaione and @ramcq on Slack, merging.

**Pull request merged**